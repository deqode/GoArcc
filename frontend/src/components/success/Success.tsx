import '../../css/bootstrap.min.css'
import '../../css/alfred.css';
import '../../css/fonts.css';
import logo from '../../images/logo.png';

function Success() {
  return (
    <div>
      <nav>
        <div className="container-fluid">
          <div className="logo">
            <a href="/" className="d-block"><img src={logo} alt="Logo!" /></a>
          </div>
        </div>
      </nav>

      <section className="content_section">
        <div className="container">
          <div className="row">
            <div className="col-md-6">
              <h1 className="page-head">Alfred has cloned your app repository</h1>
            </div>
            <div className="col-md-6 my-auto login-right">
            </div>
          </div>
        </div>
      </section>
    </div>
  )
}

export default Success
