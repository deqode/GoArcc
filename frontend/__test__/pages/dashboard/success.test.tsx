import renderer from 'react-test-renderer'
import Success, { handler } from '../../../pages/dashboard/success'

describe('dashboard/success page tests', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Success />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should return to landing page if session not available', async () => {
    const req = {
      session: { get: jest.fn(() => undefined), save: jest.fn() },
    }
    const response = await handler({ req })
    expect(response).toStrictEqual({
      redirect: {
        permanent: false,
        destination: '/',
      },
    })
  })
  it('should return user if session is available', async () => {
    const req = {
      session: { get: jest.fn(() => 'MockUser'), save: jest.fn() },
    }
    const response = await handler({ req })
    expect(response).toStrictEqual({ props: { user: 'MockUser' } })
  })
})
