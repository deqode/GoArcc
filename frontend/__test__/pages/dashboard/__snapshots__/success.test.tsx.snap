// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard/success page tests should match the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation0 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <h1
          className="MuiTypography-root MuiTypography-h3 MuiTypography-colorPrimary"
        >
          Success
        </h1>
        <br />
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
        >
          Alfred has cloned your app repository
        </h5>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <p
          className="MuiTypography-root makeStyles-text-3 MuiTypography-body1 MuiTypography-displayInline"
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;
